bot/message_processor.go

```diff
@@ -2,116 +2,139 @@ package bot
 
 import (
 	"fmt"
+	"math/rand"
 	"strings"
 
-	"github.com/18F/angrytock/slack"
+	"github.com/nlopes/slack"
 )
 
 // processMessage handles incomming messages
-func (bot *Bot) processMessage(message slackPackage.Message) {
-
-	// Check if the user is an offending user
+func (bot *Bot) processMessage(message *slack.MessageEvent) {
 	user := message.User
-	_, isViolator := bot.violatorUserMap[user]
-	// Check if user is in masterList
-	var ismasterUser bool
-	for _, masterUser := range bot.masterList {
-		if masterUser == user {
-			ismasterUser = true
-		}
+	botID := bot.Slack.GetSelfID()
+	// Handle Violators
+	userID := bot.violatorUserMap.Get(user)
+	if userID != "" {
+		bot.violatorMessage(message, user)
 	}
 
-	switch {
-	// Master user commands
-	case ismasterUser && strings.HasPrefix(message.Text, fmt.Sprintf("<@%s>", bot.Slack.ID)):
-		{
-			bot.masterMessages(message)
+	botCalled := strings.HasPrefix(
+		message.Text,
+		fmt.Sprintf("<@%s>", botID),
+	)
+	if botCalled { // Messages made directly to bot
+		switch {
+		case bot.isMasterUser(user):
+			{
+				bot.masterMessages(message)
+			}
+		default:
+			{
+				bot.niceMessage(message, user)
+			}
 		}
-	// If the user is an offending user message them, but remove them off the list
-	case isViolator:
-		{
-			bot.violatorMessage(message, user)
-		}
-		// If this is a message directed at the bot respond in a nice way
-	case strings.HasPrefix(message.Text, fmt.Sprintf("<@%s>", bot.Slack.ID)):
-		{
-			bot.niceMessage(message, user)
+	} else {
+		switch {
+		// Messages that contain the word tick
+		case strings.Contains(message.Text, " tick "):
+			{
+				bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(
+					"tock", message.Channel,
+				))
+			}
+		// Messages that references the bot will be send out 30% of the time. See: Foucault, Discipline 201
+		case strings.Contains(message.Text, fmt.Sprintf("<@%s>", botID)):
+			{
+				var returnMessage string
+				randomInt := rand.Intn(100)
+				if randomInt >= 70 {
+					returnMessage = bot.MessageRepo.Nice.GenerateMessage(user)
+				} else if randomInt <= 3 {
+					returnMessage = "'The [tockers] must never know whether [they are] looked at at any one moment;"
```

Comment:
make this into a const variable, put it above globally in this file, and just refer to it here.